autoload/go/alternate.vim

```diff
@@ -0,0 +1,36 @@
+" By default use edit to switch to the new alternate buffer
+if !exists("g:go_alternate_mode")
+	let g:go_alternate_mode = "edit"
+endif
+
+" By default create the test file if it does not exist
+if !exists("g:go_create_test_file")
+	let g:go_create_test_file  = 1
+endif
+
+" Test alternates between the implementation of code and the test code.
+function! go#alternate#Switch(cmd)
+	let file = go#alternate#Filename(fnameescape(expand("%")))
+	if g:go_create_test_file || filereadable(file)
```

Comment:
Revert this logic and return early after echoing the error message. Example:

```viml
if ... 
  echo "error message"
  return
endif

" rest of the logic here ...
```

This makes the code more readable and prevent additional if indentations